Desktop_Interface/isobuffer.h

```diff
@@ -18,62 +19,90 @@ class isoDriver;
-//Imagine it as a circular buffer, but with access functions specifically designed for isochronous data from an Xmega.
+// isoBuffer is a generic class that enables O(1) read times (!!!) on all
+// read/write operations, while maintaining a huge buffer size.
+// Imagine it as a circular buffer, but with access functions specifically
+// designed for isochronous data from an Xmega.
 
 #define CONSOLE_UPDATE_TIMER_PERIOD (ISO_PACKETS_PER_CTX * 4)
 
+// TODO: Make private what should be private
 class isoBuffer : public QWidget
 {
-    Q_OBJECT
+	Q_OBJECT
 public:
-    isoBuffer(QWidget *parent = 0, int bufferLen = 0, isoDriver *caller = 0, unsigned char channel_value = 0);
-    //Generic Functions
-    void openFile(QString newFile);
-    void writeBuffer_char(char *data, int len);
-    void writeBuffer_short(short *data, int len);
-    short *readBuffer(double sampleWindow, int numSamples, bool singleBit, double delayOffset);
-    void clearBuffer();
-    void gainBuffer(int gain_log);
-    void glitchInsert(short type);
-    double sampleConvert(short sample, int TOP, bool AC);
-    short inverseSampleConvert(double voltageLevel, int TOP, bool AC);
-    int cap_x0fromLast(double seconds, double vbot);
-    int cap_x1fromLast(double seconds, int x0, double vbot);
-    int cap_x2fromLast(double seconds, int x1, double vtop);
-    void serialManage(double baudRate, int type, UartParity parity);
-    //Generic Vars
-    QPlainTextEdit *console1, *console2;
-    bool serialAutoScroll = true;
-    unsigned char channel = 255;
-    double voltage_ref = 1.65;
-    double frontendGain = (R4 / (R3 + R4));
-    int samplesPerSecond;
-    int sampleRate_bit;
-    int bufferEnd, back = 0;
-    short *buffer, *readData = NULL;
-    uartStyleDecoder *decoder = NULL;
-    bool stopDecoding = false;
+	isoBuffer(QWidget* parent = 0, int bufferLen = 0, isoDriver* caller = 0, unsigned char channel_value = 0);
+	// TODO?: Add a destructor
+
+	//Generic Functions
+	void openFile(QString newFile);
+
+//	Basic buffer operations
+	short bufferAt(int idx) const;
+	void insertIntoBuffer(short item);
+	void clearBuffer();
+	void gainBuffer(int gain_log);
+	void glitchInsert(short type); // NO-OP
+
+// Advanced buffer operations
+	void writeBuffer_char(char* data, int len);
+	void writeBuffer_short(short* data, int len);
+
+	short* readBuffer(double sampleWindow, int numSamples, bool singleBit, double delayOffset);
+
+//	file I/O
+	bool maybeOutputSampleToFile(double convertedSample);
+	double sampleConvert(short sample, int TOP, bool AC) const;
+	short inverseSampleConvert(double voltageLevel, int TOP, bool AC) const;
+
+	int cap_x0fromLast(double seconds, double vbot);
+	int cap_x1fromLast(double seconds, int x0, double vbot);
+	int cap_x2fromLast(double seconds, int x1, double vtop);
+	void serialManage(double baudRate, int type, UartParity parity);
+
+// ---- MEMBER VARIABLES ----
```

Comment:
Prefixing these with `m_` would be good for consistency when the code is finally refactored!

Reply:
Ah, yes. Will do.